tensorflow/python/ops/check_ops.py

```diff
@@ -87,6 +87,260 @@ def _shape_and_dtype_str(tensor):
+
+def _make_assert_msg_data(sym, x, y, summarize, test_op):
+  """
+  Subroutine of _binary_assert that generates the components of the default
+  error message when running in eager mode.
+
+  Args:
+    sym: Mathematical symbol for the test to apply to pairs of tensor
+      elements, i.e. "=="
+    x, y: Inputs to the assertion after convert_to_tensor()
+    summarize: Value of the "summarize" parameter to the original assert_*
+      call; tells how many elements of each tensor to print.
+    test_op: TensorFlow op that returns a Boolean tensor with True in each
+      position where the assertion is satisfied.
+
+  Returns:
+    List of tensors and scalars that, when stringified and concatenated,
+    will produce the error message string.
+  """
+  # Prepare a message with first elements of x and y.
+  data = []
+
+  data.append('Condition x %s y did not hold.' % sym)
+
+  if summarize > 0:
+    if x.shape == y.shape and x.shape.as_list():
+      # If the shapes of x and y are the same (and not scalars),
+      # Get the values that actually differed and their indices.
+      # If shapes are different this information is more confusing
+      # than useful.
+      mask = math_ops.logical_not(test_op)
+      indices = array_ops.where(mask)
+      indices_np = indices.numpy()
+      x_vals = array_ops.boolean_mask(x, mask)
+      y_vals = array_ops.boolean_mask(y, mask)
+      num_vals = min(summarize, indices_np.shape[0])
+      data.append('Indices of first %d different values:' % num_vals)
+      data.append(indices_np[:num_vals])
+      data.append('Corresponding x values:')
+      data.append(x_vals.numpy().reshape((-1,))[:num_vals])
+      data.append('Corresponding y values:')
+      data.append(y_vals.numpy().reshape((-1,))[:num_vals])
+    
+    # reshape((-1,)) is the fastest way to get a flat array view.
+    x_np = x.numpy().reshape((-1,))
+    y_np = y.numpy().reshape((-1,))
+    x_sum = min(x_np.size, summarize)
+    y_sum = min(y_np.size, summarize)
+    data.append('First %d elements of x:' % x_sum)
+    data.append(x_np[:x_sum])
+    data.append('First %d elements of y:' % y_sum)
+    data.append(y_np[:y_sum])
+
+  return data
+
+
+def _pretty_print(data_item, summarize):
+  """
+  Format a data item for use in an error message in eager mode.
+  
+  Args:
+    data_item: One of the items in the "data" argument to an assert_*
+      function. Can be a Tensor or a scalar value.
+    summarize: How many elements to retain of each tensor-valued entry 
+      in data.
+
+  Returns an appropriate string representation of data_item
+  """
+  if isinstance(data_item, ops.Tensor):
+    arr = data_item.numpy()
+    if np.isscalar(arr):
+      # Tensor.numpy() returns a scalar for zero-dimensional tensors
+      return str(arr)
+    else:
+      flat = arr.reshape((-1,))
+      lst = [str(x) for x in flat[:summarize]]
+      if len(lst) < flat.size:
+        lst.append("...")
+      return str(lst)
+  else:
+    return str(data_item)
+
```

Comment:
Lint Error : Args section is missing, does not match arg list or is badly indented.
Expected: sym, opname, op_func, static_func, x, y, data, summarize, message, name
Found: sym, opname, op_func, static_func
Missing: x, y, data, summarize, message, name
Unexpected:

Reply:
Copied and pasted docs for `x`, `y`, `data`, `summarize`, `message`, and `name` arguments from docstring template inside `_binary_assert_doc`